### ISSUE TITLE
Invoking static method feature request

### ISSUE THREAD

**Issue by antopilo** [StudioCherno/Coral#5]
It would be quite nice to be able to invoke a static method of a class.

## Example
The proposed API would be something like so, where the assembly type exposes a new method called `invokeMethod` as well, but would invoke static methods of the assembly type.

### Managed source
```csharp

public class Engine
{
    internal static unsafe delegate*<NativeString, void> LoggerLogIcall;
    
    public static void Log()
    {
         unsafe { LoggerLogICall("HelloWorld"); }
    }
}
```

### Native source
```cpp
...
auto& assembly = loadContext.LoadAssembly("Testing.dll");
assembly.AddInternalCall("Engine", "LoggerLogIcall", reinterpret_cast<void*>(&Log));
assembly.UploadInternalCalls();

auto& engineType = assembly.GetType("Engine");

// Here, instead of creating an instance of T and invoking the method of that instance
// We could instead call invokeMethod on the type directly to invoke static methods
// auto engineInstance = engineType.CreateInstance(); 

engineType.InvokeMethod("Log"); // Using the type instead of instance here.
...

```

**Comment by peter1745** [comment]
I'll most likely get this implemented sometime tomorrow

**Comment by peter1745** [comment]
Implemented by 0d7902d
